src/linker/step.py

```diff
@@ -1,14 +1,83 @@
 from dataclasses import dataclass
-from typing import Callable
+from pathlib import Path
+from typing import Any, Callable, Dict, List
 
-from linker.utilities.data_utils import validate_dummy_output
+
+class StepInput:
+    """StepInput contains information about the input data for a step in a pipeline. It is used to bind input data to the container environment."""
+
+    def __init__(
+        self,
+        env_var: str,
+        container_dir_name: str,
+        input_filenames: List[str],
+        prev_output: bool,
+        host_filepaths: List[str] = [],
+    ):
+        self.env_var = env_var
+        self.container_dir_name = container_dir_name
+        self.input_filenames = input_filenames
+        self.prev_output = prev_output
+        ## WEIRD BUG if I don't specify the if else??
```

Comment:
In unit tests it was passing in a bunch of extraneous files from the temp directory that matched the file key. It was somehow because `host_filepaths` was in the init method but wasn't being set directly by the SCHEMA_PARAMS dictionary unpacking, I think. Otherwise, it was entirely mystical to me why it was occurring. In any case, I elected to remove `host_filepaths` from the initialization and make it so that it is only updated by the add_bindings method--I'd like to reference it in other places otherwise I'd just make it private. Maybe that's the best thing to do anyway and create an interface property.